### DISCUSSION TITLE
How to write a lambda parameter?

### DISCUSSION THREAD

**Discussion by liyihuanx** [square/kotlinpoet#1144] [Q&A] [answered]
i want to write **(e: Exception) -> Unit)? = null** as a parameter. how can i do? please help!!!!

**Comment by Egorand** [comment, marked as answer]
Start with a [`ParameterSpec.builder()`](https://square.github.io/kotlinpoet/1.x/kotlinpoet/kotlinpoet/com.squareup.kotlinpoet/-parameter-spec/-companion/builder.html) and create a [`LambdaTypeName`](https://square.github.io/kotlinpoet/1.x/kotlinpoet/kotlinpoet/com.squareup.kotlinpoet/-lambda-type-name/-companion/index.html) to represent the type. To provide the default `null` argument, use `ParameterSpec.Builder.defaultValue()`. There should be a few similar examples in the unit tests.
